lib/Elastica/Transport/HttpAsync.php

```diff
@@ -0,0 +1,103 @@
+<?php
+namespace Elastica\Transport;
+
+use Elastica\Request;
+use Elastica\Response;
+use Elastica\Util;
+
+/**
+ * Elastica HttpAsync Transport object.
+ *
+ * @author Norberto Gomez
+ */
+class HttpAsync extends AbstractTransport
+{
+    /**
+     * Http scheme.
+     *
+     * @var string Http scheme
+     */
+    protected $_scheme = 'http';
+
+    /**
+     * Curl resource to reuse.
+     *
+     * @var resource Curl resource to reuse
+     */
+    protected static $_curlConnection;
+
+    /**
+     * Makes calls to the elasticsearch server.
+     *
+     * All calls that are made to the server are done through this function
+     *
+     * @param \Elastica\Request $request
+     * @param array             $params  Host, Port, ...
+     *
+     * @return \Elastica\Response Response object
+     */
+    public function exec(Request $request, array $params)
+    {
+        $connection = $this->getConnection();
+        $baseUri = $this->_buildBaseUri($request, $connection);
+        $command = 'curl';
+        $command_params = [];
+        $command_params[] = ' -m ' . $connection->getTimeout();
+        $data = $request->getData();
+
+        $headers = [];
+        array_push($headers, sprintf('Content-Type: %s', $request->getContentType()));
```

Comment:
Maybe split the header array creation into a private method?

Reply:
Better!